extensions/jaeger/runtime/src/main/java/io/quarkus/jaeger/runtime/QuarkusJaegerTracer.java

```diff
@@ -21,13 +23,25 @@ private static Tracer tracer() {
             synchronized (QuarkusJaegerTracer.class) {
                 if (tracer == null) {
                     tracer = Configuration.fromEnv()
-                            .withMetricsFactory(new QuarkusJaegerMetricsFactory()).getTracer();
+                            .withMetricsFactory(new QuarkusJaegerMetricsFactory())
+                            .getTracerBuilder()
+                            .withScopeManager(getScopeManager())
+                            .build();
                 }
             }
         }
         return tracer;
     }
 
+    private static ScopeManager getScopeManager() {
+        ScopeManager scopeManager = new ThreadLocalScopeManager();
+        String logTraceContext = System.getProperty(LOG_TRACE_CONTEXT);
```

Comment:
I'm a bit torn on this one. Not sure using system properties is the best way to pass this value.

It's a bit different from the others because we set the others to push values to Jaeger.

Reply:
The problem is that quarkus tracer is a singleton and `tracer()` method which resolves tracer (if null) is `static` - and it's called on every tracer usage.

I would rather go with this - as it is the simplest change and conceptually follows the same approach. I can open refactoring PR which will fix https://github.com/quarkusio/quarkus/pull/6274#discussion_r360306758 and also remove passing configuration via a system property.